timelock-server/src/main/java/com/palantir/atlasdb/timelock/config/TimestampBoundStoreConfiguration.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.timelock.config;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.atlasdb.spi.KeyValueServiceConfig;
+
+@JsonSerialize(as = ImmutableTimestampBoundStoreConfiguration.class)
+@JsonDeserialize(as = ImmutableTimestampBoundStoreConfiguration.class)
+@Value.Immutable
+public abstract class TimestampBoundStoreConfiguration implements TimeLockAlgorithmConfiguration {
+    @JsonProperty("key-value-service")
+    public abstract KeyValueServiceConfig kvsConfig();
```

Comment:
What's the difference between this and the optional one added in `TimeLockInstallConfiguration`

Reply:
Its not optional if using the `timestampBoundStore` type in the timelock server config, but is optional for a general timelock install config. The timelock install config is currently constructed from the timelock server config. We should eventually get rid of it.